internal/dag/conditions_test.go

```diff
@@ -56,8 +56,16 @@ func TestPathCondition(t *testing.T) {
 			conditions: []projcontour.Condition{{
 				Prefix: "/a/",
 			}},
-			// TODO(dfc) issue 1597
-			want: &PrefixCondition{Prefix: "/a"},
+			want: &PrefixCondition{Prefix: "/a/"},
+		},
+		"trailing slash on second prefix condition": {
+			conditions: []projcontour.Condition{{
+				Prefix: "/a",
+			},
+				{
+					Prefix: "/b/",
+				}},
+			want: &PrefixCondition{Prefix: "/a/b/"},
```

Comment:
Consider adding a "doubled slashes case" like:
```Go
conditions: []projcontour.Condition{{
                                  Prefix: "////",
                         }, {
                                 Prefix: "",
                         }},
```

Reply:
I've added something like this, but note that an empty string for a prefix is not possible, the conditions validation will reject them before they get to this point.